### ISSUE TITLE
Upload sdists to PyPI

### ISSUE THREAD

**Issue by Julian** [tensorflow/tensorflow#7286]
Hi.

It appears that PyPI only contains wheels for tensorflow:

https://pypi.org/project/tensorflow/#files

Thanks for uploading those! Wheels are great.

But sdists are good to have too -- specifically, I'm not sure whether tensorflow works on PyPy (I'm about to try it out -- EDIT: now I've seen #252, which I might try to help on) but not having sdists uploaded makes that harder to try without hunting down this repository.

**Comment by girving** [comment]
@caisq Can you comment?  I'm unfamiliar with sdists.

**Comment by gunan** [comment]
I assume by sdist you assume source distribution to be built during `pip install`?
That might be tricky, as we rely on bazel. Bazel is quite heavy, and has limited os support, so it still wont work on many OSs.

**Comment by sjperkins** [comment]
https://github.com/tensorflow/tensorflow/issues/6540 may be relevant

**Comment by Julian** [comment]
#6540 appears mostly to be about eggs, which yeah I don't think tensorflow should have.

@gunan I'm still trying to understand tensorflow's logistics (I'm mostly focused on directly hitting #252 but I've certainly managed to build it via bazel by now) but my first assumption would have been that an sdist would have invoked the requisite bazel command to build itself (under the assumption that bazel is already installed, which tends to happen for sdists, you rely on the fact that the user has already done all the external non-Python work)

Basically, I would have imagined it to work the same as it does right now, just with one less manual step of having to clone the tensorflow source to get at the setup.py, ideally I'd just install bazel, then run `pip install`, and if tensorflow didn't upload a wheel compatible with my system, pip (setuptools) should invoke whatever tool tensorflow uses to build one (and if it succeeds, great).

**Comment by girving** [comment]
Unfortunately I think this is too hard at present given the complexities of bazel.  Closing for now.
